frontend/src/scenes/data-warehouse/editor/QueryWindow.tsx

```diff
@@ -147,12 +148,25 @@ function RunButton(): JSX.Element {
     const { runQuery } = useActions(multitabEditorLogic)
     const { cancelQuery } = useActions(dataNodeLogic)
     const { responseLoading } = useValues(dataNodeLogic)
-    const { metadata, queryInput } = useValues(multitabEditorLogic)
+    const { metadata, queryInput, isSourceQueryLastRun } = useValues(multitabEditorLogic)
 
     const isUsingIndices = metadata?.isUsingIndices === 'yes'
-    const tooltipContent = !isUsingIndices
-        ? 'This query is not using indices optimally, which may result in slower performance.'
-        : undefined
+
+    const [iconColor, tooltipContent] = useMemo(() => {
+        if (isSourceQueryLastRun) {
+            return ['var(--primary)', 'No changes to run']
+        }
+
+        if (!metadata || isUsingIndices || queryInput.trim().length === 0) {
+            return ['var(--success)', 'New changes to run']
+        }
+
+        const tooltipContent = !isUsingIndices
+            ? 'This query is not using indices optimally, which may result in slower performance.'
+            : undefined
+
+        return ['var(--warning)', tooltipContent]
```

Comment:
Nit: Not sure you really need the variable here and I think it would read a bit better to just inline:
```tsx
return ['var(--warning)', !isUsingIndices
    ? 'This query is not using indices optimally, which may result in slower performance.'
    : undefined]
```

Reply:
going to merge for now so CI isn't cranky for another hour